src/main/scala/com/microsoft/hyperspace/index/sources/delta/DeltaLakeFileBasedSource.scala

```diff
@@ -160,18 +168,25 @@ class DeltaLakeFileBasedSource(private val spark: SparkSession) extends FileBase
   }
 
   /**
-   * Constructs the basePath for the given [[FileIndex]].
+   * Constructs the basePath for the given [[LogicalPlan]].
    *
-   * @param location Partitioned data location.
+   * @param logicalPlan Logical plan to extract the base path from.
    * @return basePath to read the given partitioned location.
+   *         Some(Some(path)) => The location of the given plan is supported
+   *                             and partition is specified.
+   *         Some(None) => The location of the given plan is supported
+   *                       but is un-partitioned.
+   *         None => The location of the given plan is not supported.
    */
-  override def partitionBasePath(location: FileIndex): Option[Option[String]] = {
-    location match {
-      case d: TahoeLogFileIndex if d.partitionSchema.nonEmpty =>
-        Some(Some(d.path.toString))
-      case _: TahoeLogFileIndex => Some(None)
-      case _ =>
-        None
+  override def partitionBasePath(logicalPlan: LogicalPlan): Option[Option[String]] = {
+    getLogicalRelation(logicalPlan).map(_.relation).flatMap {
+      case HadoopFsRelation(d: TahoeLogFileIndex, _, _, _, _, _) =>
```

Comment:
The same pattern is being used in different methods, so I think an extractor would help?

Reply:
👍